core-web/libs/portlets/edit-ema/portlet/src/lib/store/dot-uve.store.ts

```diff
@@ -0,0 +1,127 @@
+import { patchState, signalStore, withComputed, withMethods, withState } from '@ngrx/signals';
+
+import { computed } from '@angular/core';
+
+import { withEditor } from './features/editor/withEditor';
+import { withLayout } from './features/layout/withLayout';
+import { withLoad } from './features/load/withLoad';
+import { ShellProps, UVEState } from './models';
+
+import { COMMON_ERRORS } from '../shared/consts';
+import { UVE_STATUS } from '../shared/enums';
+import { sanitizeURL } from '../utils';
+
+const initialState: UVEState = {
+    isEnterprise: false,
+    languages: [],
+    pageAPIResponse: null,
+    currentUser: null,
+    experiment: null,
+    error: null,
+    params: null,
+    status: UVE_STATUS.LOADING,
+    isTraditionalPage: true,
+    canEditPage: false,
+    pageIsLocked: true
+};
+
+export const UVEStore = signalStore(
+    withState<UVEState>(initialState),
+    withComputed((store) => {
+        return {
+            $shellProps: computed<ShellProps>(() => {
+                const pageAPIResponse = store.pageAPIResponse();
+
+                const currentUrl = '/' + sanitizeURL(pageAPIResponse?.page.pageURI);
+
+                const requestHostName = !store.isTraditionalPage()
+                    ? store.params()?.clientHost
+                    : window.location.origin;
+
+                const page = pageAPIResponse?.page;
+                const templateDrawed = pageAPIResponse?.template.drawed;
+
+                const isLayoutDisabled = !page?.canEdit || !templateDrawed;
+
+                const languageId = pageAPIResponse?.viewAs.language.id;
+                const languages = store.languages();
+                const errorCode = store.error();
+
+                return {
+                    canRead: page?.canRead,
+                    error: errorCode
```

Comment:
same here, move the logic to a utils, and use here something like `getError(errorCode)` it is more easy move all this logic to a util and create a unit test easily just to a function, not to the whole store

Reply:
Done